libminifi/src/core/yaml/YamlConnectionParser.cpp

```diff
@@ -29,24 +29,45 @@ namespace yaml {
 // This is no longer needed in c++17
 constexpr const char* YamlConnectionParser::CONFIG_YAML_CONNECTIONS_KEY;
 
+void YamlConnectionParser::addNewRelationshipToConnection(const std::string& relationship_name, const std::shared_ptr<minifi::Connection>& connection) const {
+  core::Relationship relationship(relationship_name, "");
+  logger_->log_debug("parseConnection: relationship => [%s]", relationship_name);
+  connection->addRelationship(std::move(relationship));
+}
+
+void YamlConnectionParser::addFunnelRelationshipToConnection(const std::shared_ptr<minifi::Connection>& connection) const {
+  utils::Identifier srcUUID;
+  try {
+    srcUUID = getSourceUUIDFromYaml();
+  } catch(const std::exception&) {
+    return;
+  }
+  auto processor = parent_->findProcessorById(srcUUID);
+  if (!processor) {
+    return;
+  }
+  if (std::dynamic_pointer_cast<minifi::core::Funnel>(processor)) {
```

Comment:
If `Funnel` can be `final`, then this could be done more efficiently with `typeid` comparison, because it doesn't have to travel the inheritance graph.

Reply:
Changed in a76b8c6b77f48f19dbc805a0a26a9f55e015d40e